Lib/python/pyhead.swg

```diff
@@ -87,3 +87,17 @@ SWIG_Python_str_FromChar(const char *c)
 #define PyDescr_NAME(x) (((PyDescrObject *)(x))->d_name)
 #define Py_hash_t long
 #endif
+
+#ifdef Py_LIMITED_API
+# define PyTuple_GET_ITEM PyTuple_GetItem
+/* Note that PyTuple_SetItem() has different semantics from PyTuple_SET_ITEM as it decref's the original tuple item, so in general they cannot be used
+  interchangeably. However in SWIG-generated code PyTuple_SET_ITEM is only used with newly initialized tuples without any items and for them this does work. */
+# define PyTuple_SET_ITEM PyTuple_SetItem
```

Comment:
This seems dangerous since existing user typemap code might use `PyTuple_SET_ITEM` in other ways and so break in likely obscure ways if they try it with the limited API.  There's also a risk of someone using it in a future SWIG version in other ways.

Maybe it should redirect through `SWIG_PyTyple_ADD_ITEM()` or something like that?  (`ADD` to make it clear it works when adding a new item rather than a general set that could replace.)

Reply:
This would result in more changes and I'm not sure if it's worth making them. Also, as this only affects the use case which doesn't exist today, I think we could definitely merge this first and replace the calls to `PyTuple_SET_ITEM()` with whatever later.

Reply:
Actually, I had a look at the Python C API docs for this, which say (my emphasis):

> Like `PyTuple_SetItem()`, but does no error checking, and **should only be used to fill in brand new tuples**.

So using it on something that isn't a brand new tuple is documented as something not to do, and in the light of that just mapping it like above seems OK - any user typemap broken by this would be dubious code.